enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,307 @@
+---
+title: microshift-supportability-tools
+authors:
+  - pacevedom
+reviewers:
+  - "@copejon, MicroShift contributor"
+  - "@fzdarsky, MicroShift architect"
+  - "@ggiguash, MicroShift contributor"
+  - "@oglok, MicroShift contributor"
+  - "@pmtk, MicroShift contributor"
+  - "@TurboTurtle, SOS contributor"
+  - "@jrthms, product experience liaison"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2023-01-17
+last-updated: 2023-01-17
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-389
+see-also:
+  - "/enhancements/oc/must-gather.md"
+---
+
+# MicroShift supportability tools
+
+## Summary
+This enhancement proposes the addition of supportability tooling to MicroShift.
+
+To debug issues in the cluster we need a single command that any user or
+administrator is able to run to gather all the required information to diagnose
+and solve problems.
+
+## Motivation
+Most readers will be familiar with `must-gather` for OpenShift clusters and
+what it achieves. For MicroShift we need a tool with the same principles and
+goals, which is to gather every bit of information from the cluster to be able
+to diagnose and fix issues.
+
+### User Stories
+
+* As a MicroShift administrator, I want to be able to get a report about current
+  cluster status.
+* As a MicroShift developer, I want to be able to diagnose issues with a detailed
+  debug report.
+
+### Goals
+* Provide a single tool to automate the report creation.
+
+### Non-Goals
+N/A
+
+## Proposal
+While `must-gather` tool is a widely known and used for debugging clusters, it
+may not be the most suitable for MicroShift use cases, and requires deep
+modifications. The existing `must-gather` implementation:
+1. Needs many OpenShift APIs present, like `clusteroperators`.
```

Comment:
No, that's not true. The must-gather uses a collection script to collect the data. When the pod is created it calls the must gather entry point - https://github.com/openshift/must-gather/blob/master/collection-scripts/gather - This calls sub collection scripts which are collecting additional data. You can simply create a custom gather script and collect only information you need.

Reply:
You are right, but for the pod to be created you need a new container image that we want to avoid in MicroShift.
Besides, we would need to fork our own must-gather, right?